ide/coqide/preferences.ml

```diff
@@ -310,45 +310,48 @@ let automatic_tactics =
 let cmd_print =
   new preference ~name:["cmd_print"] ~init:"lpr" ~repr:Repr.(string)
 
-let attach_modifiers (pref : string preference) prefix =
+let attach_modifiers (pref : string preference) ?(filter = Fun.const true) prefix =
   let cb mds =
     let mds = str_to_mod_list mds in
     let change ~path ~key ~modi ~changed =
-      if CString.is_sub prefix path 0 then
+      if CString.is_prefix prefix path && filter path then
         ignore (GtkData.AccelMap.change_entry ~key ~modi:mds ~replace:true path)
     in
     GtkData.AccelMap.foreach change
   in
-  pref#connect#changed ~callback:cb
+  ignore (pref#connect#changed ~callback:cb)
+
+let select_arch m m_osx =
+  if Coq_config.arch = "Darwin" then m_osx else m
```

Comment:
Seems too mac-centric.  Maybe should be a bit more general.

OTOH, I tend to agree with Pierre-Marie that remapping keys is a losing game, perhaps better left to each user at this point.

Reply:
> Seems too mac-centric. Maybe should be a bit more general.

This is not my code, I just moved the lines for this to be less messy. On macOS, things are very different:
- “Command” (Super) seems to have the role of “Ctrl” (for common hotkeys);
- “Option” (Alt) seems to have the role of “Ctrl” (for moving by word) or “AltGr”;
- “Ctrl” seems to have the role of “Super”.

Thus I think distinguishing between macOS and other systems like this is fully justified. GTK offers “Primary” as a way to designate macOS's Command and other systems’ Ctrl at the same time (even web browsers don’t do that); but sometimes, this is not enough.

> OTOH, I tend to agree with Pierre-Marie that remapping keys is a losing game, perhaps better left to each user at this point.

Many users don’t bother browsing offered settings. The initial configuration is very important. But yeah, developing something to ease changing hotkeys could be a good idea.

Reply:
Unfortunately, if a key is not bound in `coqide.keys`, it gets the default setting.  If we change the default, then any user who had the default suddenly has the new value.  Not so friendly.

Reply:
Sure, I haven’t planned to do it often. In this case, I think people will be able to understand once that default key bindings were a bit messy.